File: 2109 adding-spaces-to-a-string/solution.go
func addSpaces(s string, spaces []int) string {
    slen, splen := len(s), len(spaces)
    ans := make([]byte, slen+splen)
    idx, sp := 0, 0

    for i := 0; i < slen; i++ {
        if sp < splen && i == spaces[sp] {
            ans[idx] = ' '
            idx++
            sp++
        }
        ans[idx] = s[i]
        idx++
    }
    return string(ans)
}

File: 2109 adding-spaces-to-a-string/solution.cs
public class Solution {
    public string AddSpaces(string s, int[] spaces) {
        int slen = s.Length,splen = spaces.Length;
        char[] ans = new char[slen+splen];
        int idx = 0,sp = 0;
        for(int i=0;i<slen;i++) {
            if(sp<splen && i==spaces[sp]) {
                ans[idx++] = ' ';
                sp++;
            }
            ans[idx++] = s[i];
        }
        return new string(ans);
    }
}
